src/action-ros-ci.ts

```diff
@@ -555,10 +555,13 @@ done`;
 	// being built.
 	let repoFullName = process.env.GITHUB_REPOSITORY as string;
 	if (github.context.payload.pull_request) {
-		repoFullName = github.context.payload.pull_request.head.repo.full_name;
+		repoFullName = github.context.payload.pull_request.base.repo.full_name;
```

Comment:
If I understand correctly, the git commit for the "PR merge" commit that we are trying to check out is only available on the target repo (i.e., the original repo against which the PR was opened), correct?

Reply:
correct. But even more critically, the token which GH generates automatically in the target repo is not valid in the source repo (which is not public).

Reply:
@christophebedard since there is no one maintaining vcstool, and this tool is maintained.
Would you consider dropping usage of vctools and clone repos in a different way?
Might be easier then trying to fix it in 2 repos

Reply:
> But even more critically, the token which GH generates automatically in the target repo is not valid in the source repo (which is not public).

Got it.

> since there is no one maintaining vcstool, and this tool is maintained.

The goal is for OSRF to get maintainer access eventually, but that's going to take some time.

> Would you consider dropping usage of vctools and clone repos in a different way?
> Might be easier then trying to fix it in 2 repos

For the repo being tested, yes, that's a good idea. Running plain old `git` commands instead of using `vcstool` would just be simpler here. Would you be willing to make this change? I wouldn't change anything about the `vcs-repo-file-url` option (for other repos), though.